var/spack/repos/builtin/packages/symengine/package.py

```diff
@@ -0,0 +1,118 @@
+from spack import *
+
+
+class Symengine(Package):
+    """SymEngine is a fast symbolic manipulation library, written in C++."""
+
+    homepage = "http://sympy.org"
+    url      = "https://github.com/symengine/symengine/archive/v0.2.0.tar.gz"
+
+    version('0.2.0', '45401561add36a13c1f0b0c5f8d7422d')
+    version('0.1.0', '41ad7daed61fc5a77c285eb6c7303425')
+    version('develop', git='https://github.com/symengine/symengine.git')
+
+    # Libraries to add to spack:
+    # Flint: www.flintlib.org
+
+    # variant('bfd',          default=False,
+    #         description='Compile with BFD library')  # Error: Requires linkh?
+    # variant('flint',        default=False,
+    #         description='Compile with Flint library')
+    variant('mpc',          default=True,
+            description='Compile with MPC library')
+    variant('mpfr',         default=True,
+            description='Compile with MPFR library')
+    variant('piranha',      default=False,
+            description='Compile with Piranha library')
+    variant('openmp',       default=False,
+            description='Enable OpenMP support')
+    variant('shared',       default=True,
+            description='Enables the build of shared libraries')
+
+    # Build dependencies
+    depends_on('cmake',    type='build')
+
+    # Other dependencies
+    depends_on('gmp')
+    depends_on('binutils', when='+bfd')
+    depends_on('mpc',      when='+mpc')
+    depends_on('mpfr',     when='+mpfc')
+    depends_on('piranha',  when='+piranha')
+    # depends_on('flint',    when='+flint~piranha')
+
+    def install(self, spec, prefix):
+        options = []
+        options.extend(std_cmake_args)
+
+        # CMAKE_BUILD_TYPE should be Debug | Release
+        for word in options[:]:
+            if word.startswith('-DCMAKE_BUILD_TYPE'):
+                options.remove(word)
+
+        # See https://github.com/symengine/symengine/blob/master/README.md
+        # for build options
+        options.extend([
+            '-DCMAKE_BUILD_TYPE=Release',
+            '-DWITH_SYMENGINE_RCP:BOOL=ON',
+            '-DWITH_SYMENGINE_THREAD_SAFE:BOOL=ON',
+            '-DBUILD_TESTS:BOOL=ON',
+            '-DBUILD_BENCHMARKS:BOOL=ON',
+            # '-DWITH_BFD:BOOL=%s' % (
+            #     'ON' if '+bfd' in spec else 'OFF'),
+            '-DWITH_MPC:BOOL=%s' % (
+                'ON' if '+mpc' in spec else 'OFF'),
+            '-DWITH_MPFR:BOOL=%s' % (
+                'ON' if '+mpfr' in spec else 'OFF'),
+            '-DINTEGER_CLASS:STRING=gmp',
+            '-DWITH_OPENMP:BOOL=%s' % (
+                'ON' if '+openmp' in spec else 'OFF'),
+            '-DBUILD_SHARED_LIBS:BOOL=%s' % (
+                'ON' if '+shared' in spec else 'OFF'),
+        ])
+
+        if '+piranha' in spec:
+            options.extend([
+                '-DWITH_PIRANHA:BOOL=ON',
+                '-DINTEGER_CLASS:STRING=piranha'
+            ])
+        elif '+flint' in spec:
```

Comment:
You can find the packages here, http://mirrors.mit.edu/sage/spkg/upstream/flint/index.html or in any of the mirrors of sage.